src/praisonai/praisonai/cli/features/serve.py

```
@@ -303,12 +399,41 @@ async def invoke_agents(request: Request, query_data: AgentQuery = None):
+                    # Create a simple PraisonAgent instance
+                    agent = PraisonAgent(
+                        name=agent_data.get('role', agent_name),
+                        instructions=f"{agent_data.get('goal', '')}. {agent_data.get('backstory', '')}".strip(),
+                        # Add tools if they exist in the YAML
+                        tools=agent_data.get('tools', [])
+                    )
+                    
+                    # Register the agent
+                    register_agent_func(agent_name, agent)
+                    
+                except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)